sql/datadict.cc

```diff
@@ -46,25 +46,26 @@ static int read_string(File file, uchar**to, size_t length)
 
   engine_name is a LEX_CSTRING, where engine_name->str must point to
   a buffer of at least NAME_CHAR_LEN+1 bytes.
-  If engine_name is 0, then the function will only test if the file is a
-  view or not
+  If both engine_name and is_versioned are 0, then the function will
+  only test if the file is a view or not
 
   @param[out] is_sequence  1 if table is a SEQUENCE, 0 otherwise
+  @param[out] is_versioned 1 if table is  versioned, 0 otherwise
 
   @retval  TABLE_TYPE_UNKNOWN   error  - file can't be opened
   @retval  TABLE_TYPE_NORMAL    table
   @retval  TABLE_TYPE_SEQUENCE  sequence table
   @retval  TABLE_TYPE_VIEW      view
 */
 
-Table_type dd_frm_type(THD *thd, char *path, LEX_CSTRING *engine_name,
-                       bool *is_sequence)
+enum Table_type dd_frm_type(THD *thd, char *path, LEX_CSTRING *engine_name, bool *is_sequence, bool *is_versioned)
```

Comment:
Default `is_versioned= NULL` will save couple of hunks.

Reply:
There's a trade between saving couple of hunks, and authenticity: i mean, there are another two `out` parameters, `engine_name` and `is_sequence` which are made as required arguments. So it looks like previous authors prefer not to use default arguments (neither do I, btw).
Maybe, if you insist on minimizing the patch, it is better to make all of them (i mean, those last three) default-valued then?

Reply:
That would not minimize the patch. Though in fact I would prefer it that way. But you I guess there are more people that really prefer not to use default values. So probably you should keep it this way.